lib/abort-signal.js

```diff
@@ -24,6 +24,10 @@ function addSignal (self, signal) {
   } else {
     self[kSignal].addListener('abort', self[kListener])
   }
+
+  if (signal.aborted) {
+    self[kListener]()
```

Comment:
You _probably_ want to not add the signal and just call the listener? Is the signal correctly removed in this case?

Reply:
Fixed